### PR TITLE
VUE-209 Initial pass at KvAccordionItem component

### PR DESCRIPTION
**Pull request by ryan-ludwig**
Mostly code stolen from InfoPanel.vue. Props and events modeled after the `details` element.

image

image

### CODE REVIEW

File: src/components/Kv/KvAccordion.vue [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+				 * Contains an object with a boolean 'open' property of the current open state of the accordion
+				 * @event toggle
+				 * @type {Event}
+				 */
+				this.$emit('toggle', { open: this.isOpen });
+			}
+		},
+		expand() {
+			if (!this.disabled) {
+				this.isOpen = true;
```
Comment (mcstover) [inline src/components/Kv/KvAccordion.vue:103]: We need at least a generic tracking event on the clicks for opening and closing these.

**Comment by mcstover** [comment]
The screenshots look nice!

File: src/components/Kv/KvAccordion.vue (hunk 1 above)
Comment (ryan-ludwig) [reply src/components/Kv/KvAccordion.vue:103]: I was thinking that tracking would be handled by the parent of these components, by hooking into the 'toggle' event.

Based on what we have now, 
```

// FrequentlyAskedQuestions.vue
<kv-accordion @toggle="trackFAQInteraction" ...>
    <template v-slot:title><h3>{{title}}</h3></template>
    <div>lorem ipsum...</div>
</kv-accordion>

trackFAQInteraction(event) {
    if (event.open) {
        this.$kvTrackEvent(this.analyticsCategory, 'click-faq-expand', this.title);
    }
}


// LoanDetailsPanel.vue
<div>
<kv-accordion @toggle="trackLoanInteraction" :id="panelId">
    <template v-slot:title><h3>{{title}}</h3></template>
    <div>lorem ipsum...</div>
</div>

trackLoanInteraction(event) {
    this.$emit('track-interaction', {
        interactionType: `${event.open ? 'open' : 'close'}-${this.panelId}-panel`,    
    }
}
```

**Comment by emuvente** [comment]
This looks great! Though I'm wondering if calling this an accordion might cause some confusion, as accordion components in many other libraries usually involve multiple collapsed elements and ensure that only one of those elements is open at a time

File: src/components/Kv/KvAccordion.vue (hunk 1 above)
Comment (mcstover) [reply src/components/Kv/KvAccordion.vue:103]: Ok, I suppose that works for now.

**Comment by mcstover** [comment]
@ryan-ludwig @emuvente maybe we could go with KvAccordianItem thought I'm not certain this could be used as is in a dynamic group of content such as our FAQs.

**Comment by ryan-ludwig** [comment]
@emuvente, @mcstover  Good call, I was wondering about that too. I like the idea of naming this `KvAccordionItem`. Gives us room to make `KvAccordion` in the future which could keep track of all it's child `KvAccordionItem`s and only allow one open at a time.

**Comment by ryan-ludwig** [comment]
@mcstover For converting FAQs, it'd look something like

Existing (FrequentlyAskedQuestions.vue):
```
<kv-expandable-question
    v-for="(question, index) in faqsContentful"
    :key="index"
    :title="question.name"
    :content="convertFromRichTextToHtml(question.richText)"
    :id="question.name | changeCase('paramCase')"
/>
```
Future:
```
<kv-accordion
    v-for="(question, index) in faqsContentful"
    :key="index"
    :id="question.name | changeCase('paramCase')"
    @toggle="handleTracking"
>
    <template v-slot:header><h3>{{question.name}}</h3></template>
    <div v-html="convertFromRichTextToHtml(question.richText)" />
</kv-accordion>
```

**Review by mcstover — APPROVED** [review]

**Review by emuvente — APPROVED** [review]
:+1:

**Review by BoulderBrains — APPROVED** [review]
👍

**Review by BoulderBrains — APPROVED** [review]
👍

**Review by eddieferrer — APPROVED** [review]

**Pull request merged**